src/org/omegat/core/data/RealProject.java

```diff
@@ -1851,18 +1856,20 @@ public boolean existEntryInProject(EntryKey key) {
 
     void setOnlineMode() {
         if (!isOnlineMode) {
+            isOnlineMode = true;
+            preparedStatus = PreparedStatus.NONE;
             Log.logInfoRB("VCS_ONLINE");
             Core.getMainWindow().displayWarningRB("VCS_ONLINE", "VCS_OFFLINE");
         }
-        isOnlineMode = true;
```

Comment:
I would actually feel better if `isOnlineMode` and `preparedStatus` were set unconditionally here and in `setOfflineMode()`.